oss-internship-2020/libpng/examples/example1.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "../sandboxed.h"  // NOLINT(build/include)
+#include "../tests/libpng.h"  // NOLINT(build/include)
+#include "sandboxed_api/vars.h"
+
+absl::Status LibPNGMain(const std::string& infile, const std::string& outfile) {
+  LibPNGSapiSandbox sandbox;
+  sandbox.AddFile(infile);
+  sandbox.AddFile(outfile);
+
+  SAPI_RETURN_IF_ERROR(sandbox.Init());
+  LibPNGApi api(&sandbox);
+
+  sapi::v::Struct<png_image> image;
+  sapi::v::ConstCStr infile_var(infile.c_str());
+  sapi::v::ConstCStr outfile_var(outfile.c_str());
+
+  image.mutable_data()->version = PNG_IMAGE_VERSION;
+
+  SAPI_ASSIGN_OR_RETURN(
+      int result, api.png_image_begin_read_from_file(image.PtrBoth(),
+                                                     infile_var.PtrBefore()));
+  if (!result) {
+    return absl::InternalError(
+        absl::StrCat("begin read error: ", image.mutable_data()->message));
+  }
+
+  image.mutable_data()->format = PNG_FORMAT_RGBA;
+
+  auto buffer = malloc(PNG_IMAGE_SIZE(*image.mutable_data()));
```

Comment:
What's the use of this buffer here? It is only malloc'ed and freed (and not used in the meanwhile).
Also if the allocation fails it will return okstatus in line 48, is that intended?